config/routes.rb

```diff
@@ -1,5 +1,7 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-
-  resources :posts
+  resources :posts do
+    resources :comments do
+    end
```

Comment:
Can you explain what is happening here with a nested route block? When you do `resources :posts` my understanding is that this builds loads of default routes like `../posts`, `.../posts/new` . So for example does this nested block mean there is now valid url/route `...posts/new/comments/new` ?

Reply:
nested resources is to show the path dependency unless there’s a  post made (post path) one can not comment(/go to comment url) so for each post we can access comment resources but without a post made one can’t check comments functionality. (that’s my understanding from the workshop)

Reply:
Okay nice that makes sense thank you

Reply:
The posts & comments would then be nested under the user routes as they need to be dependent by it...

Reply:
I don't think that would necessarily be the case in terms of routes. You're right in saying if there are no users there can be no posts, but I think the point of this is to create a dependent relationship between posts and their comments.

If we were introducing account pages, where you could go to users page and see only their posts, maybe we would want to do something similar. But since at the moment all we have is just a feed of all posts ever made - regardless of who the current logged in user is, or who the user who posted it was, I don't think we need to create a dependancy between users and posts. 

Reading this helped me understand https://guides.rubyonrails.org/routing.html#nested-resources

Reply:
@suhani-zaman or @yousufmohamed17  - did you have to add all the the `has_many`, `belongs_to` etc in the model files in light of this nested resources?

Reply:
The nested route doesn't create any dependencies. It just creates RESTful routes that branch off the parent so instead of /comments you get /posts/:post_id/comments. The `has_many` and `belongs_to` creates the dependency between the tables.

Reply:
What do you mean by this:

> The `has_many` and `belongs_to` creates the dependency between the tables.

The tables are already essentially dependent on each other because of the foreign key constraint? By dependent in that context I mean say there us a user with id 1 and a 3 posts in the posts table made by user 1, then because the foreign key 1 is referenced in the posts table, we can't delete user 1 from the users table, until all its posts in the posts table are deleted. Same goes posts & comments. I think I'm unclear on what else a dependancy would be at the table level

Reply:
If you only used a foreign key you'd have to do exactly what you said and delete the posts then the user. But with the `has_many`/`belongs_to` associations you can tell rails to `@user.destroy` a particular user and it will delete the user and all the posts/comments that belong to the user. So it makes the code simpler. There is a good guide on this here that I read: https://guides.rubyonrails.org/association_basics.html

Reply:
Okay lovely that makes sense -thank you ✨